### ISSUE TITLE
false positive - csharpsquid:S2325

### ISSUE THREAD

**Issue by jmecosta** [SonarSource/sonarlint-vs#404]
hi,

we have this code:

```csharp
interface InterfaceX
{
    bool DoSomething(int componentNumber, string componentName, string attributeFile);
}


class xpto : InterfaceX
{

        [DllImport("somenative.dll", CallingConvention = CallingConvention.Cdecl)]
        [SuppressMessage("Microsoft.StyleCop.CSharp.NamingRules", "*", Justification = "Native naming conventions")]
        private static extern bool somenativemenber(int componentNumber, string componentName, string attributeFile);

        public bool DoSomething(int componentNumber, string componentName, string attributeFile)
        {
            return somenativemenber(componentNumber, componentName, attributeFile);
        }
}
```

S2325 is raised against public bool DoSomething(int componentNumber, string componentName, string attributeFile)

**Comment by tvsonar** [comment]
@jmecosta Thanks for coming back with this issue. We experienced quite a few false positives with this rule, so we have dropped it in SonarLint 1.1.0. 

See the release notes here: http://vs.sonarlint.org/rules/index.html#version=1.1.0, and the conversation that lead to this decision: https://github.com/SonarSource/sonarlint-vs/issues/65.

Can you confirm that you are running an older version of SonarLint?

**Comment by jmecosta** [comment]
oh, totally missed that defect... 
hum.... ive checked the dll version in 1.1.0 and its the same as the one i am using. 1.2.0
i still use the java analyser  4.1 on server....

**Comment by tvsonar** [comment]
Can you run me through how do you check the version and which dll are you checking? Are you using the SonarLint VSIX, the SonarLint Nuget package, or the MSBuild SonarQube Runner?

**Comment by jmecosta** [comment]
None, using the java sonar runner. Checking the version was just right
click on dll and see version.

**Comment by dbolkensteyn** [comment]
yep that's expected @jmecosta - the C# plugin 4.1 embeds an older version of the SonarLint VS assembly. We need to re-ship the plugin each time we want to include a newer release.

**Comment by dbolkensteyn** [comment]
Duplicates #65

**Comment by jmecosta** [comment]
Yep that's what I thought I will rebuild the plugin with new version to
confirm its working
